includes/class-wc-payments-webhook-processing-service.php

```diff
@@ -154,6 +154,12 @@ public function process( array $event_body ) {
 			case 'charge.refunded':
 				$this->process_webhook_refund_triggered_externally( $event_body );
 				break;
+			case 'charge.failed':
+				// TODO: Implement this. During manual tests with a physical card reader, we found that this event can also be triggered:
+				// https://dashboard.stripe.com/test/connect/accounts/acct_1KBG7y2HCBWb6DsL/payments/ch_3PicUz2HCBWb6DsL1hY7VS3F
+				// https://dashboard.stripe.com/test/connect/accounts/acct_1KBG7y2HCBWb6DsL/payments/ch_3PicVR2HCBWb6DsL1sXHsjfa
+				// Can we just call $this->process_webhook_payment_intent_failed( $event_body ); ?
```

Comment:
I think you cannot since the event body could be different.  I cannot open the listed events, but I think that the payment_intent failed is sent here too. Is that correct?

Reply:
Yes, the payment_intent failed is also sent (I did not see it initially because I was looking at the charge details and not the payment_intent). It also sends a payment_intent.canceled.

We should be good with adding the card present to the actionable methods, but Jirka tested it (details in the p2 thread) and it did not seem to work. We need to keep investigating why